src/main/js/reducers/grades.js

```diff
@@ -1,23 +1,18 @@
 import {
-  ADD_COURSEGPA, SELECT_SORTBYGPA,
+  ADD_GRADES,
 } from '../actions/types';
 
 const initialState = {
   map: {},
-  sort: false,
 };
 
 export default (state = initialState, action) => {
   switch (action.type) {
-    case ADD_COURSEGPA:
+    case ADD_GRADES:
       return {
         ...state,
-        map: action.gpa,
-      };
-    case SELECT_SORTBYGPA:
-      return {
-        ...state,
-        sort: action.sort,
+        map: action.payload,
```

Comment:
If there's an error, we wouldn't want the map to be an error object, this would break components since it relies on this value. You'd need to check the action if it has an error and set map accordingly.

Reply:
Done.